controllers/GroupController.js

```diff
@@ -0,0 +1,130 @@
+import GroupModel from '../models/Group.js'
+
+export const getAll = async (req, res) => {
+    try {
+        const groups = await GroupModel.find({ user: req.userId }).exec();
+        res.json(groups);
+    } catch (error) {
+        console.log(error);
+        res.status(500).json({
+            message: 'Не вдалося отримати групи',
+        })
+    }
+}
+
+export const getOne = async (req, res) => {
+    try {
+        const groupId = req.params.id;
+        GroupModel.findOne({
+            _id: groupId,
+            user: req.userId
+        }, (error, doc) => {
+            if (error) {
+                console.log(error);
+                return res.status(500).json({
+                    message: "Не вдалося повернути групу" 
+                });
+            }   
+            if (!doc) {
+                return res.status(404).json({
+                    message: "Група не знайдена"
+                });
+            }
+            res.json(doc);
```

Comment:
Replace the use of callbacks with async/await syntax for `findOne` and `findOneAndUpdate` methods. This will improve code readability and provide convenient error handling:
```suggestion
export const getOne = async (req, res) => {
    try {
        const groupId = req.params.id;
        const group = await GroupModel.findOne({
            _id: groupId,
            user: req.userId
        });

        if (!group) {
            return res.status(404).json({
                message: "Група не знайдена"
            });
        }

        res.json(group);
```